File: 251002/Samoilenko/publisher/src/main/java/com/example/discussion/exception/GlobalExceptionHandler.java
package com.example.discussion.exception;

import com.example.discussion.dto.ErrorResponse;
import jakarta.persistence.EntityExistsException;
import jakarta.persistence.EntityNotFoundException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.MethodArgumentNotValidException;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;


@ControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler({EntityExistsException.class, MethodArgumentNotValidException.class, DuplicateLoginException.class, EntityNotFoundException.class})
    public ResponseEntity<ErrorResponse> handleExceptions(Exception ex) {
        ErrorResponse errorResponse = new ErrorResponse(ex.getMessage());
        return new ResponseEntity<>(errorResponse, HttpStatus.FORBIDDEN);
    }
}

File: 251003/Pankratiev/rest/discussion/src/main/kotlin/bsuir/dc/discussion/controller/GlobalExceptionHandler.kt
package bsuir.dc.discussion.controller

import bsuir.dc.discussion.dto.to.ExceptionResponseTo
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(NoSuchElementException::class)
    fun handleNoSuchElementException(ex: NoSuchElementException): ResponseEntity<ExceptionResponseTo> {
        return ResponseEntity(ExceptionResponseTo(ex.message), HttpStatus.NOT_FOUND)
    }
}
